### PR TITLE
change apt to update cache

### LINKED ISSUE #95: Ansible playbooks fail to install packages
http://maas-integration-ci.internal:8080/view/ansible-playbook/job/ansible-playbook-tests/51/consoleText

Seems the cache needs to be updated before installing anything.

### PR DESCRIPTION
**Pull request by SK1Y101**
Should™ fix #95

### CODE REVIEW

File: roles/maas_postgres/tasks/install_postgres.yml [hunk 1]
```diff
@@ -1,18 +1,11 @@
 ---
 - name: "Install psycopg2 For Configuration"
```
Comment (sparkiegeek) [inline roles/maas_postgres/tasks/install_postgres.yml:2]:
```suggestion
- name: "Install dependencies for PostgreSQL configuration"
```

File: roles/maas_postgres/tasks/teardown.yaml [hunk 2]
```diff
@@ -9,16 +9,10 @@
...
-- name: "Uninstall acl"
-  ansible.builtin.apt:
-    name: acl
-    state: absent
-
-- name: "Uninstall psycopg2"
-  ansible.builtin.apt:
-    name: "python3-psycopg2"
-    state: absent
+    update_cache: true
```
Comment (sparkiegeek) [inline roles/maas_postgres/tasks/teardown.yaml:18]: shouldn't need to `apt-get update` for removing - this just operates on the local system

File: roles/maas_proxy/tasks/teardown.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 - name: "Uninstall HAProxy"
   ansible.builtin.apt:
-    name: haproxy
+    name: "haproxy"
     state: absent
     update_cache: true
```
Comment (sparkiegeek) [inline roles/maas_proxy/tasks/teardown.yaml:6]:
```suggestion
```

same here, no need to update on removal

File: roles/maas_rack_controller/tasks/teardown.yaml [hunk 4]
```diff
@@ -7,12 +7,9 @@
...
-    purge: true
-    state: absent
-
-- name: Uninstall MAAS - deb dependencies
-  ansible.builtin.apt:
-    name: "chrony"
+    name:
+      - "maas*"
+      - "chrony"
+    update_cache: true
```
Comment (sparkiegeek) [inline roles/maas_rack_controller/tasks/teardown.yaml:13]:
```suggestion
```

File: roles/maas_region_controller/tasks/teardown.yaml [hunk 5]
```diff
@@ -11,6 +11,7 @@
     name: "maas*"
     purge: true
     state: absent
+    update_cache: true
```
Comment (sparkiegeek) [inline roles/maas_region_controller/tasks/teardown.yaml:14]:
```suggestion
```

File: teardown.yaml [hunk 6]
```diff
@@ -56,3 +56,4 @@
   - name: "Remove unused packages from the cache"
     ansible.builtin.apt:
       autoclean: true
+      update_cache: true
```
Comment (sparkiegeek) [inline teardown.yaml:59]:
```suggestion
```

**Review by sparkiegeek — CHANGES_REQUESTED** [review]

File: roles/maas_postgres/tasks/teardown.yaml (hunk 2 above)
Comment (SK1Y101) [reply roles/maas_postgres/tasks/teardown.yaml:18]: Fair point, I think I added update_cache to a few removal tasks

**Review by sparkiegeek — APPROVED** [review]

File: roles/maas_rack_controller/tasks/teardown.yaml [hunk 7]
```diff
@@ -7,12 +7,8 @@
...
-    name: "maas*"
-    purge: true
-    state: absent
-
-- name: Uninstall MAAS - deb dependencies
-  ansible.builtin.apt:
-    name: "chrony"
+    name:
+      - "maas*"
+      - "chrony"
```
Comment (sparkiegeek) [inline roles/maas_rack_controller/tasks/teardown.yaml:12]:
```suggestion
```

File: roles/maas_rack_controller/tasks/upgrade_maas.yaml [hunk 8]
```diff
@@ -8,15 +8,12 @@
...
   ansible.builtin.apt:
-    name: chrony
-    state: latest
-  when: (maas_installation_type | lower == 'deb')
-
-- name: Install MAAS Rack Controller
-  ansible.builtin.apt:
-    name: "maas-rack-controller"
+    name:
+      - "chrony"
```
Comment (sparkiegeek) [inline roles/maas_rack_controller/tasks/upgrade_maas.yaml:14]:
```suggestion
```

File: roles/maas_region_controller/tasks/update_maas.yaml [hunk 9]
```diff
@@ -10,9 +10,12 @@
 
 - name: Update MAAS - Deb
   ansible.builtin.apt:
-    name: maas-region-api
+    name:
+      - "chrony"
```
Comment (sparkiegeek) [inline roles/maas_region_controller/tasks/update_maas.yaml:14]:
```suggestion
```

File: roles/maas_region_controller/tasks/teardown.yaml [hunk 10]
```diff
@@ -8,7 +8,9 @@
 
 - name: Uninstall MAAS - Deb
   ansible.builtin.apt:
-    name: "maas*"
+    name:
+      - "maas*"
+      - "chrony"
```
Comment (sparkiegeek) [inline roles/maas_region_controller/tasks/teardown.yaml:13]:
```suggestion
```

File: roles/maas_postgres/tasks/teardown.yaml [hunk 11]
```diff
@@ -9,16 +9,9 @@
 
 - name: "Uninstall Postgres"
```
Comment (sparkiegeek) [inline roles/maas_postgres/tasks/teardown.yaml:10]:
```suggestion
- name: "Uninstall PostgreSQL and configuration dependencies"
```

**Review by sparkiegeek — APPROVED** [review]

**Pull request merged**